.github/workflows/python-ci.yml

```diff
@@ -16,19 +16,28 @@ jobs:
       uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
+
+    - name: Install pipenv
+      run: |
+        pip install -U pipenv
+    
+    - id: cache-pipenv
+      uses: actions/cache@v2
+      with:
+        path: ~/.local/share/virtualenvs
+        key: ${{ runner.os }}-pipenv-${{ hashFiles('**/Pipfile.lock') }}
+
     - name: Install dependencies
-      env:
-        PIP_INDEX_URL: https://pypi.pacificclimate.org/simple
+      if: steps.cache-pipenv.outputs.cache-hit != 'true'
```

Comment:
Nice. Suggest changing `name` of this step to something like "Install dependencies if changed" and/or adding a comment to that effect.